### PR TITLE
Fix multiple API mistakes made in define removal

### PR DESCRIPTION
**Pull request by chamons**
- A number of small correction in foundation and mapkit
- The [quicklook PR](https://github.com/xamarin/xamarin-macios/pull/14491) was completed reverted and redone
- Instead of trying to remove duplication (which can't be done since they are in different namespaces), I just took the originals and slapped attributes [NoMac] or NoEverythingElse on everything.

In the future we can do a better quicklook job, but this is good enough, and we have time pressure due to branching. 

API diff run locally. Only green additions, and each of those was confirmed to be either valid addition from a define removal or unrelated to a define removal PR.

### CODE REVIEW

**Comment by chamons** [comment]
I validated these changes by doing the following:

- Syncing to 891610b53, which was before any of the changes landed
`* 891610b53 - (HEAD) [nnyeah] Warn on IConvertible (#14586) (4 days ago)<Steve Hawley>`
- Full build
- `cd tools/apidff && make update-refs1
- Sync after my changes
- Full build
- `cd tools/apidiff && make`
- open api-diff.html and review each section

File: src/quicklook.cs [hunk 1]
```diff
@@ -127,6 +125,7 @@ interface QLPreviewControllerDelegate {
 		[Export ("previewController:shouldOpenURL:forPreviewItem:"), DelegateName ("QLOpenUrl"), DefaultValue (false)]
 		bool ShouldOpenUrl (QLPreviewController controller, NSUrl url, [Protocolize] QLPreviewItem item);
 
+#if !MONOMAC
```
Comment (tj-devel709) [inline src/quicklook.cs:128]: This one is already inside a !MONOMAC from line 49, but I understand this is a quick PR!

**Review by tj-devel709 — APPROVED** [review]

**Comment by chamons** [comment]
This is Xamarin.iOS.dll, Xamarin.TVOS.dll, Microsoft.iOS.dll, Microsoft.tvOS.dll, 

image

Xamarin.Mac.dll and Microsoft.macOS.dll

image

Microsoft.MacCatalyst.dll has the stuff everyone else has but then a bunch of EventKitUI additions from [here](https://github.com/xamarin/xamarin-macios/commit/1c4a64c0c44cd83d0fe6480858a5f5b13e7e7e0d)

image

File: src/quicklook.cs (hunk 1 above)
Comment (chamons) [reply src/quicklook.cs:128]: Yep, I decided to revert and make a few changes as humanly possible to get it fixed without losing the define/attribute quality. 

I'll leave refactoring to a future day.

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by dalexsoto** [comment]
@sudo backport release/6.0.3xx-rc1

**Comment by dalexsoto** [comment]
Oh my bad it is not needed, those got in after rc1 was branched, phew!

**Pull request merged**